File: frontend/src/types/persona.ts
export type CommunicationStyle = 'Analytical' | 'Narrative' | 'Terse' | 'Verbose';

export interface Persona {
  id: string;
  created_time: string;
  persona_type: string;
  primary_use_case: string;
  communication_style: CommunicationStyle;
  goals: string;
  constraints: string;
  speaker_name: string;
  persona_summary: string;
}

File: backend/models/persona.py
from typing import Literal

from pydantic import BaseModel


CommunicationStyle = Literal["Analytical", "Narrative", "Terse", "Verbose"]


class PersonaCreate(BaseModel):
    persona_type: str
    primary_use_case: str
    communication_style: CommunicationStyle
    goals: str
    constraints: str


class Persona(PersonaCreate):
    id: str
    created_time: str
